src/structures/VoiceState.js

```diff
@@ -196,6 +209,73 @@ class VoiceState extends Base {
     return true;
   }
 
+  /**
+   * Toggles the request to speak in the channel.
+   * Only applicable for stage channels and for the client's own voice state.
+   * @param {boolean} request Whether or not the client is requesting to become a speaker.
+   * @example
+   * // Making the client request to speak in a stage channel (raise its hand)
+   * guild.me.voice.setRequestToSpeak(true);
+   * @example
+   * // Making the client cancel a request to speak
+   * guild.me.voice.setRequestToSpeak(false);
+   */
+  async setRequestToSpeak(request) {
+    const channel = this.channel;
+    if (channel?.type !== 'stage') throw new Error('VOICE_NOT_STAGE_CHANNEL');
+
+    if (this.client.user.id !== this.id) throw new Error('VOICE_STATE_NOT_OWN');
+
+    const member = this.member;
+    const hasRequestToSpeakPermission = member?.permissionsIn(channel).has(Permissions.FLAGS.REQUEST_TO_SPEAK);
+
+    if (!hasRequestToSpeakPermission) throw new Error('VOICE_NEED_REQUEST_TO_SPEAK');
```

Comment:
We don't check for permissions before making http requests elsewhere, is this something exceptional?